scripts/gha/pr_file_commenter.py

```diff
@@ -0,0 +1,238 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Take warnings/error lines from stdin and turn them into PR file comments.
+
+Usage:
+python pr_file_commenter.py -t github_token -p pr_number
+       -T comment_tag [-r git_repo_url] [-C curl_command]
+       [-P comment_prefix] [-S comment_suffix] [-d base_directory]
+       < COMMENT_LINES
+
+COMMENT_LINES should be a series of lines with the following format:
+path/to/first_filename:line_number: comment text
+ optional comment text continuation
+path/to/second_filename:line_number: comment text
+
+If -r is unspecified, uses the current repo.
+"""
+
+import argparse
+from html import escape
+import json
+import os
+import re
+import subprocess
+import sys
+import time
+from unidiff import PatchSet
+import urllib.parse
+
+def main():
+  # This script performs a number of steps:
+  #
+  # 1. Get the PR's diff to find the list of affected files and lines in the PR.
+  #
+  # 2. Get the list of comments to post. Remove duplicates, and then
+  #    omit any comment that doesn't fall in the affected lines.
+  #
+  # 3. Delete any prior comments posted by previous runs.
+  #
+  # 4. Post any comments that fall within the range of the PR's diff.
+
+  args = parse_cmdline_args()
+  HIDDEN_COMMENT_TAG = '<hidden value="%s"></hidden>' % args.comment_tag
+  if args.repo is None:
+      args.repo=subprocess.check_output(['git', 'config', '--get', 'remote.origin.url']).decode('utf-8').rstrip('\n').lower()
+      if args.verbose:
+        print('autodetected repo: %s' % args.repo)
+  if not args.repo.startswith('https://github.com/'):
+      print('Error, only https://github.com/ repositories are allowed.')
+      exit(2)
+  (repo_owner, repo_name) = re.match(r'https://github\.com/([^/]+)/([^/.]+)', args.repo).groups()
+
+  # Get the head commit for the pull request.
+  # GET /repos/{owner}/{repo}/pulls/{pull_number}
+  request_url = 'https://api.github.com/repos/%s/%s/pulls/%s' % (repo_owner, repo_name, args.pr_number)
+  header = 'Accept: application/vnd.github.VERSION.json'
+  pr_data = json.loads(subprocess.check_output(
+      [args.curl,
+       '-s', '-X', 'GET',
+       '-H', 'Accept: application/vnd.github.v3+json',
+       '-H', 'Authorization: token %s' % args.token,
+       request_url
+      ] + ([] if not args.verbose else ['-v'])).decode('utf-8').rstrip('\n'))
+
+  commit_sha = pr_data['head']['sha']
+  if args.verbose:
+    print('Commit sha:', commit_sha)
+
+  # Get the diff for the pull request.
+  # GET /repos/{owner}/{repo}/pulls/{pull_number}
+  request_url = 'https://api.github.com/repos/%s/%s/pulls/%s' % (repo_owner, repo_name, args.pr_number)
+  header = 'Accept: application/vnd.github.VERSION.diff'
```

Comment:
Is it worth puting this "delete previous comment" and "get commit number of branch" into a common import?

Reply:
I think I will in the future, probably when I refactor lint_commenter to use this.

Reply:
(Also, planning to update these to use python _requests_ rather than calling curl.)